### DISCUSSION TITLE
Automatic heartbeat start with activation not working?

### DISCUSSION THREAD

**Discussion by nschloe** [keygen-sh/community#64] [Issue]
[The Feb 2023 changelog](https://keygen.sh/changelog/) says:

> When a machine which requires a heartbeat is activated, its heartbeat is now automatically started at time of activation.

This doesn't seem to work for me. I activate the license with
```python
requests.post(
    f"https://api.keygen.sh/v1/accounts/{account_id}/machines",
    headers={
        "Authorization": f"License {license_key}",
        "Content-Type": "application/vnd.api+json",
        "Accept": "application/vnd.api+json",
    },
    data=json.dumps(
        {
            "data": {
                "type": "machines",
                "attributes": {
                    "fingerprint": machine_fingerprint,
                },
                "relationships": {
                    "license": {
                        "data": {
                            "type": "licenses",
                            "id": license_id,
                        },
                    },
                },
            },
        },
    ),
    timeout=10,
).json()
```
and then validate with

```python
requests.post(
    f"https://api.keygen.sh/v1/accounts/{account_id}/licenses/actions/validate-key",
    headers={
        "Content-Type": "application/vnd.api+json",
        "Accept": "application/vnd.api+json",
    },
    data=json.dumps(
        {
            "meta": {
                "key": license_key,
                "scope": machine_fingerprint,
            }
        }
    ),
    timeout=10,
)
```

Result:
```python
{
    // ... 
    'meta': {
        'ts': '2023-03-02T13:42:34.705Z',
        'valid': False,
        'detail': 'machine heartbeat is required',
        'scope': {'fingerprint': '...'},
        'constant': 'HEARTBEAT_NOT_STARTED'
    }
}
```

**Comment by ezekg** [comment]
The license's policy likely needs to enable [`requireHeartbeat`](https://keygen.sh/docs/api/policies/#policies-object-attrs-requireHeartbeat). In addition, the policy's [`heartbeatBasis`](https://keygen.sh/docs/api/policies/#policies-object-attrs-heartbeatBasis) needs to be set to `FROM_CREATION`. Older accounts, created before this feature was introduced in v1.3, have their policies default to a heartbeat basis of `FROM_FIRST_PING`, which matches previous behavior.

Your pinned API version can be changed via [your account's settings](https://app.keygen.sh/settings).

**Reply by nschloe** [reply]
> In addition, the policy's [heartbeatBasis](https://keygen.sh/docs/api/policies/#policies-object-attrs-heartbeatBasis) needs to be set to FROM_CREATION

That was the issue. Thanks!
